### DISCUSSION TITLE
Constants and expressions support

### DISCUSSION THREAD

**Discussion by m-kru** [esynr3z/corsair#107] [Questions]
Does Corsair support constants and expressions? For example, I would like to define two module constants `X` and `Y`, and then I would like to calculate different values utilizing these constants in the expressions.

**Comment by esynr3z** [comment]
It is not possible currently. Adding such feature to be fully operational when declararive formats like YAML/JSON are used is quite a tricky thing. 
BTW, there is also quite similar proposal #50. 

Anyway, I have plans to add `.py` file as in input format for register map, so your code can simply provide some RegisterMap object or class. Something similar to the way you can provide your custom Generator to corsair.

By using that format you'll get all the power of Python to make your map filled by calculated things and derived values.

**Reply by v0lker** [reply]
doing it this way sounds reasonable and it solves the issue in the general case.

what i proposed in #50 (now #108) / PR #51 is much simpler but only works when the evaluation of those expressions can be delayed (in this case the base address).
there seems little benefit in implementing both, so, if you already aim to solve the general case, i'd say #108 should probably be dropped...?
